pilot/pkg/xds/filters/filters.go

```diff
@@ -138,3 +163,50 @@ var (
 	mtlsHTTP11ALPN = []string{"istio-http/1.1", "istio"}
 	mtlsHTTP2ALPN  = []string{"istio-h2", "istio"}
 )
+
+func buildHTTPMxFilter() *hcm.HttpFilter {
+	vmConfig := &v3.PluginConfig_VmConfig{
+		VmConfig: &v3.VmConfig{
+			Runtime: "envoy.wasm.runtime.null",
+			Code: &core.AsyncDataSource{Specifier: &core.AsyncDataSource_Local{
+				Local: &core.DataSource{
+					Specifier: &core.DataSource_InlineString{
+						InlineString: "envoy.wasm.metadata_exchange",
+					},
+				},
+			}},
+		},
+	}
+
+	if features.EnableWasm {
```

Comment:
nit: suggestion to put `var vmConfig *v3.PluginConfig_VmConfig` first, and then provide two values in if/else blocks.

Reply:
yes, that is much more clear to read